### PR TITLE
📖 Updates quick start docs for CAPV

### PR DESCRIPTION
**Pull request by srm09**
**What this PR does / why we need it**:
This patch updates the quick start docs for CAPV.

**Which issue(s) this PR fixes**:
n/a

### CODE REVIEW

File: docs/book/src/user/quick-start.md [hunk 1]
```diff
@@ -604,12 +604,14 @@ export VSPHERE_RESOURCE_POOL="*/Resources"
 export VSPHERE_FOLDER="vm"
 # The VM template to use for your VMs
 export VSPHERE_TEMPLATE="ubuntu-1804-kube-v1.17.3"
-# The VM template to use for the HAProxy load balancer of the management cluster
-export VSPHERE_HAPROXY_TEMPLATE="capv-haproxy-v0.6.0-rc.2"
 # The public ssh authorized key on all machines
 export VSPHERE_SSH_AUTHORIZED_KEY="ssh-rsa AAAAB3N..."
-
-clusterctl init --infrastructure vsphere
```
Comment (fabriziopandini) [inline docs/book/src/user/quick-start.md:612]: Q: Are we sure we want to drop this instruction for vSphere (the users can miss this step if not instructed to do so....)

Comment (fabriziopandini) [reply docs/book/src/user/quick-start.md:612]: never mind, just noticed this is in the generate template section

Comment (srm09) [reply docs/book/src/user/quick-start.md:612]: This instruction is already a part of the variable setup for vSphere. Hence removed it from here.
https://github.com/kubernetes-sigs/cluster-api/blame/main/docs/book/src/user/quick-start.md#L330

Comment (srm09) [reply docs/book/src/user/quick-start.md:612]: Page didn't refresh, missed your second comment. :+1:

**Review by fabriziopandini — COMMENTED** [review]
/approve

**Pull request merged**